resources/config_external_image_storage.rb

```diff
@@ -0,0 +1,61 @@
+#
+# Cookbook Name:: grafana
+# Resource:: config_external_image_storage
+#
+# Copyright 2018, Sous Chefs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Configures the installed grafana instance
+
+property  :instance_name,           String,         name_property: true
+property  :conf_directory,          String,         default: '/etc/grafana'
+property  :config_file,             String,         default: lazy { ::File.join(conf_directory, 'grafana.ini') }
+property  :storage_provider,        String,         required: true, default: 's3'
+property  :region,                  String,         required: true, default: 'us-east-1'
+property  :bucket,                  String,         required: true
+property  :bucket_url,              String,         required: false
+property  :path,                    String,         required: false
+property  :access_key,              String,         required: false
```

Comment:
Not familiar here, why would these not be required?

Reply:
@Xorima the path is optional. I modifed the code so that atleast we need to set either bucket/bucket_url. As far as it goes with the access_key and secret_key they are optional. We can rely on the IAM role of the ec2 instance that is running on Grafana. That's why I set those as optional